### PR TITLE
✨ Source Bing Ads: add new stream AgeGenderAudienceReport

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolving https://github.com/airbytehq/airbyte/issues/17380

## How

- remove old caching => use standard `request_cache` instead
- add new reports: `AgeGenderAudienceReport`

## Recommended reading order
1. `source.py`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`7d730d3ac4`](https://github.com/airbytehq/airbyte/pull/31712/commits/7d730d3ac4a1dc795f3baa80d443caa4c95c2d97)) - ❌

⏲️  Total pipeline duration: 56.87s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ❌       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698140953/7d730d3ac4a1dc795f3baa80d443caa4c95c2d97/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7d730d3ac4a1dc795f3baa80d443caa4c95c2d97)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`a76f249368`](https://github.com/airbytehq/airbyte/pull/31712/commits/a76f249368b45b340576ff1a0109754e8c0f6076)) - ❌

⏲️  Total pipeline duration: 48.65s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ❌       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698141262/a76f249368b45b340576ff1a0109754e8c0f6076/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a76f249368b45b340576ff1a0109754e8c0f6076)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`0a61de1d46`](https://github.com/airbytehq/airbyte/pull/31712/commits/0a61de1d46903eb90b56bf43c6eb131bc69ca193)) - ❌

⏲️  Total pipeline duration: 01mn11s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ❌       |
| Check our base image is used                                   | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698141976/0a61de1d46903eb90b56bf43c6eb131bc69ca193/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:0a61de1d46903eb90b56bf43c6eb131bc69ca193)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`b7c53f4ab6`](https://github.com/airbytehq/airbyte/pull/31712/commits/b7c53f4ab6aaf4575b80e8d3a3a8bd371453604f)) - ❌

⏲️  Total pipeline duration: 48.02s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ❌       |
| Check our base image is used                                   | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698143755/b7c53f4ab6aaf4575b80e8d3a3a8bd371453604f/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b7c53f4ab6aaf4575b80e8d3a3a8bd371453604f)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`b8e5c36c79`](https://github.com/airbytehq/airbyte/pull/31712/commits/b8e5c36c7946b2684189b11a92009a8bcb977ce0)) - ❌

⏲️  Total pipeline duration: 27mn22s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ❌       |
| Check our base image is used                                   | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698145461/b8e5c36c7946b2684189b11a92009a8bcb977ce0/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b8e5c36c7946b2684189b11a92009a8bcb977ce0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by bazarnov** [comment]
Creds issues, while CAT run, expired token.

**Review by alafanechere — APPROVED** [review]
Approving the bypass reason addition.
Please make sure to merge only when you have 🟢 tests.

File: airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ acceptance_tests:
             bypass_reason: "Hourly reports are disabled, because sync is too long"
           - name: geographic_performance_report_hourly
             bypass_reason: "Hourly reports are disabled, because sync is too long"
+          - name: age_gender_audience_report_hourly
+            bypass_reason: "Hourly reports are disabled, because sync is too long"
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml:38]: Can you mention how long is it? We can tweak the `timeout_seconds` value.

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`7682351f7b`](https://github.com/airbytehq/airbyte/pull/31712/commits/7682351f7bfec3d9762838882e8e8f3604371a28)) - ❌

⏲️  Total pipeline duration: 42.60s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ❌       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698153541/7682351f7bfec3d9762838882e8e8f3604371a28/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7682351f7bfec3d9762838882e8e8f3604371a28)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

File: airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml]: description updated; hourly data fetched is limited to 180 days (or even maybe 14 days: found different numbers on 3rd-party sites)

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`7edf7d28a3`](https://github.com/airbytehq/airbyte/pull/31712/commits/7edf7d28a3771a834f4e52f1eca3f5709cb2ecb0)) - ❌

⏲️  Total pipeline duration: 27mn15s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ✅       |
| Check our base image is used                                   | ✅       |
| Code format checks                                             | ❌       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698154813/7edf7d28a3771a834f4e52f1eca3f5709cb2ecb0/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7edf7d28a3771a834f4e52f1eca3f5709cb2ecb0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-bing-ads test report (commit [`b2c833bf39`](https://github.com/airbytehq/airbyte/pull/31712/commits/b2c833bf39f7a3743da62ae516a5cd33a0ed520e)) - ✅

⏲️  Total pipeline duration: 15mn26s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-bing-ads docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ✅       |
| Check our base image is used                                   | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-bing-ads                          | ✅       |
| Connector version semver check                                 | ✅       |
| Connector version increment check                              | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-bing-ads-new-reports/1698159804/b2c833bf39f7a3743da62ae516a5cd33a0ed520e/source-bing-ads/1.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b2c833bf39f7a3743da62ae516a5cd33a0ed520e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-bing-ads test
```

**Pull request merged**